gravitee-am-test/specs/gateway/user-registration.spec.ts

```diff
@@ -0,0 +1,320 @@
+/*
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fetch from "cross-fetch";
+import {clearEmails, getLastEmail} from "@utils-commands/email-commands";
+import * as faker from 'faker';
+import {jest, afterAll, beforeAll, expect} from "@jest/globals";
+import {requestAdminAccessToken} from "@management-commands/token-management-commands";
+import {createDomain, deleteDomain, startDomain, waitFor, waitForDomainSync} from "@management-commands/domain-management-commands";
+import { createUser, getAllUsers, getUser, listUsers } from "@management-commands/user-management-commands";
+import { extractXsrfToken, extractXsrfTokenAndActionResponse, getWellKnownOpenIdConfiguration, performFormPost, performGet } from "@gateway-commands/oauth-oidc-commands";
+import { createIdp, getAllIdps } from "@management-commands/idp-management-commands";
+import { createApplication, updateApplication } from "@management-commands/application-management-commands";
+import cheerio from "cheerio";
+import { fake } from "faker";
+import { decodeJwt } from "@utils-commands/jwt";
+
+global.fetch = fetch;
+
+const jdbc = process.env.GRAVITEE_MANAGEMENT_TYPE;
+
+let domain;
+let accessToken;
+let defaultIdp;
+let customIdp;
+let application;
+let clientId;
+let openIdConfiguration;
+
+jest.setTimeout(200000)
+
+beforeAll(async () => {
```

Comment:
the file name should contain "jest.spec.ts" to align with rest of the test names. This comment is not relates to this line. Rather it is relates to the test file name :) . You missed the "jest" word

Suggestion: Also noticed the file is not formatted. Would be helpful for future developer if the file is formatted.